storage/filesystem/internal/dotgit/dotgit_test.go

```diff
@@ -216,7 +260,92 @@ func (s *SuiteDotGit) TestPackfile(c *C) {
 	}
 }
 
+func (s *SuiteDotGit) TestObjectfiles(c *C) {
+	objectfiles := func(d *DotGit) (fs.FS, []core.Hash, error) {
+		return d.Objectfiles()
+	}
+	for _, test := range [...]struct {
+		fixture string
+		fn      getObjectsHashesFn
+		err     string // error regexp
+	}{
+		{
+			fixture: "unpacked",
+			fn:      objectfiles,
+		},
+		{
+			fixture: "unpacked-dummy",
+			fn:      objectfiles,
+		}, {
+			fixture: "empty",
+			fn:      objectfiles,
+			err:     ".* no such file or directory",
+		}, {
+			fixture: "no-packfile-no-idx",
+			fn:      objectfiles,
+		},
+	} {
+		com := Commentf("fixture = %s", test.fixture)
+
+		fix, dir := s.newFixtureDir(c, test.fixture)
+
+		_, hashes, err := test.fn(dir)
+
+		if test.err != "" {
+			c.Assert(err, ErrorMatches, test.err, com)
+		} else {
+			c.Assert(err, IsNil, com)
+			c.Assert(len(hashes), Equals, len(fix.objectfiles), com)
+			for i, hash := range hashes {
+				c.Assert(hash.String(), Matches, fix.objectfiles[i].hash, com)
+			}
+		}
+	}
+}
+
+func (s *SuiteDotGit) TestObjectfile(c *C) {
+	objectfile := func(d *DotGit, h core.Hash) (fs.FS, string, error) {
+		return d.Objectfile(h)
+	}
+	for _, test := range [...]struct {
+		fixture string
+		fn      getObjectPathFn
+		err     string // error regexp
```

Comment:
You have defined a new error for objectfile not found (`ErrObjfileNotFound`), will it be better to compare the var instead of using a regexp?